docs/decisions/0001-shellcheck.rst

```diff
@@ -0,0 +1,79 @@
+
+In almost all repositories, Open edX Python code is checked for common mistakes using pylint. Many repositories go further and use static analysis tools such as pycodestyle, pydocstyle, mypy, black, and isort. In constrast, even in spite of the known difficulty of correct shell scripting, Open edX shell scripts are *not* currently checked with any static analysis tools. Therefore, as shell scripts continue to be used in new Open edX contexts like `translations tooling <https://github.com/openedx/openedx-atlas>`_ and `static asset tooling <https://github.com/openedx/edx-platform/pull/31790>`_, we would like to identify a verification tool and encourage Open edX developers to use it.
+
+ShellCheck
+==========
+
+`ShellCheck <https://shellcheck.net>`_ seems very suitable:
+
+* It is well-known, with over 31.5k stars on GitHub. Pylint, in comparison, has 4.5k.
+* Its error codes, including suggested resolution steps, are `documented on its wiki <https://www.shellcheck.net/wiki/SC1000>`_.
+* It can be installed in one line from various `common package managers <https://github.com/koalaman/shellcheck#user-content-installing>`_ including apt and homebrew.
+* It can be integrated with `common editors <https://github.com/koalaman/shellcheck#user-content-in-your-editor>`_ including Vim and VSCode.
+* As of the writing of this ADR, it `seems to be actively maintained <https://github.com/koalaman/shellcheck/commits/master>`_.
+
+Finally, as a personal anecdote: ShellCheck has been extremely helpful to this ADR's author in `rewriting the edx-platform asset build in Bash <https://github.com/openedx/edx-platform/pull/31791>`_. In addition to catching simple mistakes like typos, ShellCheck's detailed error pages have provided understanding of important shell concepts such as `globbing and word splitting <https://www.shellcheck.net/wiki/SC2086>`_.
+
+Decision
+********
+
+Open edX developers are encouraged to use ShellCheck as part of the CI suite for repositories containing shell scripts.
+
+Consequences
+************
+
+* A configurable ShellCheck workflow template will be added to this repository. This will let developers add ShellChecking to their repository's PRs with a couple clicks.
+
+  * In order to ensure deterministic builds, the template will pin ShellCheck to a specific version (v0.9.0 at time of writing). The maintainer of this repository will strive to update the template's version pin over time. Maintainers of repositories using ShellCheck may stick with the default provided in this repo or can override to a different version.  Those that override it  will be encouraged to update their pin over time. We expect that not all maintainers will do this, leading to lag & drift between versions; we expect that this will be OK, given that Bash itself is essentially feature-complete, so any newly-introduced ShellCheck warnings are not expected to be critical.
```

Comment:
```suggestion
  * To support deterministic builds, the template will allow repositories to independently pin ShellCheck to a specific version. We will encourage maintainers of these repositories to update their pin over time. We expect that this will lead to some lag & drift between versions; however, we expect that this will be OK, given that Bash itself is essentially feature-complete, so any newly-introduced ShellCheck warnings are not expected to be critical.
```

@feanil @timmc-edx I'd like to reword this bullet point a bit to make it less specific. That should give us the flexibility over in https://github.com/openedx/.github/pull/64 to decide what combination of template, reusable workflow, and third-party action we want. Let me know if you have any issue with this wording, otherwise I'll merge tomorrow.

Reply:
This seems good to me.

Reply:
Yep, looks good.